### PR TITLE
feat(utils): added uslices

### PR DESCRIPTION
**Pull request by keruch**
Added `uslices` and `utypes` packages.

### CODE REVIEW

File: utils/uslices/slices.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+	var l int
+	for _, slice := range s {
+		l += len(slice)
+	}
+	r := make(S, 0, l)
+	for _, slice := range s {
+		r = append(r, slice...)
+	}
+	return r
+}
```
Comment (danwt) [inline utils/uslices/slices.go]: already in stdlib slices.concat?

File: utils/utypes/converters.go [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package utypes
+
+// StringToType converts a string to a custom type T, where T can be either a string or a byte slice.
+// May be used as a parameter for uslices.Map.
+func StringToType[T ~string | ~[]byte](s string) T { return T(s) }
```
Comment (danwt) [inline utils/utypes/converters.go]: `FromString` name is better

File: utils/utypes/converters.go [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+
+// TypeToString converts a custom type T, where T can be either a string or a byte slice, back to a string.
+// May be used as a parameter for uslices.Map.
+func TypeToString[T ~string | ~[]byte](t T) string { return string(t) }
```
Comment (danwt) [inline utils/utypes/converters.go]: `ToString` name is better

Comment (danwt) [inline utils/utypes/converters.go]: I think what you have here is more suited to a `ustring` package

**Review by danwt — CHANGES_REQUESTED** [review]

File: utils/uslices/slices.go (hunk 1 above)
Comment (keruch) [reply utils/uslices/slices.go]: oh yeah you are right, deleted it

**Review by danwt — APPROVED** [review]

**Pull request merged**